### DISCUSSION TITLE
Azure carbon impact

### DISCUSSION THREAD

**Discussion by AmmaRed562** [electricitymaps/electricitymaps-contrib#5764] [Q&A]
Hello everyone,
I am currently working on a report that calculates the carbon impact of my Azure databricks clusters. I started using the official [emissions-impact-dashboard](https://www.microsoft.com/en-us/sustainability/emissions-impact-dashboard)  given by Azure but wanted to confirme the numbers using electricitymaps. Is that possible ?
For example if i have a [Standard_D3_v2](https://azureprice.net/vm/Standard_D3_v2)  in The Netherlands what impact will it have vs running it in the UK ?
Thanks for your help.

**Comment by VIKTORVAV99** [comment]
Hi!

I don't know what numbers you have access to in the Azure dashboard or how they calculate things but you could use the API for this and monitor the difference between the two zones. Just note that we use data on a grid level and Azure might be using more granular data (like including on-location backup power plants (including battery) or egress/ingress from other zones or countries).

It also seems like Microsoft is using Scope 3 emissions for this particular dashboard which include a lot more than just electricity. (Electricity Maps is scope 2)
![image](https://github.com/electricitymaps/electricitymaps-contrib/assets/30777521/2e6e44d5-fc50-4de4-bf9d-51f5bb936282)

**Reply by AmmaRed562** [reply]
Thank you for your reply. 

For example if I can calculate the overall energy consumption of my cluster and get something like 60 kWh for a month, what dataset should I use to calculate it's CO2 emissions ?

**Reply by VIKTORVAV99** [reply]
To get he most accurate readings you would have to calculate the consumption per hour for any given hour and then get the corresponding emission factor or emissions for that hour and then compute a weighted average for the month.

The reason for this is I assume your usage is not constant and there are some fluctuations so you might use more power when the sun is up (if it is located in the same region your users are) and get a lower carbon intensity than the average or you might use more at night (when there is no solar power, like overnight data processing) and get a higher carbon intensity than the average.

Our [API](https://api-portal.electricitymaps.com/home) only supports hourly values, but our [data portal](https://www.electricitymaps.com/data-portal) has hourly, daily, monthly and yearly but only has data for 2021 and 2022 for a subset of all zones right now.
